crates/cast/bin/cmd/constructor_args.rs

```diff
@@ -0,0 +1,104 @@
+use alloy_dyn_abi::DynSolType;
+use alloy_primitives::{Address, Bytes};
+use clap::{command, Parser};
+use eyre::{eyre, OptionExt, Result};
+use foundry_block_explorers::Client;
+use foundry_cli::{
+    opts::{EtherscanOpts, RpcOpts},
+    utils,
+};
+use foundry_config::Config;
+
+use super::{
+    creation_code::fetch_creation_code,
+    interface::{fetch_abi_from_etherscan, load_abi_from_file},
+};
+
+/// CLI arguments for `cast creation-args`.
+#[derive(Parser)]
+pub struct ConstructorArgsArgs {
+    /// An Ethereum address, for which the bytecode will be fetched.
+    contract: Address,
+
+    /// Path to file containing the contract's JSON ABI. It's necessary if the target contract is
+    /// not verified on Etherscan
+    #[arg(long)]
+    abi_path: Option<String>,
+
+    #[command(flatten)]
+    etherscan: EtherscanOpts,
+
+    #[command(flatten)]
+    rpc: RpcOpts,
+}
+
+impl ConstructorArgsArgs {
+    pub async fn run(self) -> Result<()> {
+        let Self { contract, etherscan, rpc, abi_path } = self;
+
+        let config = Config::from(&etherscan);
+        let chain = config.chain.unwrap_or_default();
+        let api_key = config.get_etherscan_api_key(Some(chain)).unwrap_or_default();
+        let client = Client::new(chain, api_key)?;
+
+        let config = Config::from(&rpc);
+        let provider = utils::get_provider(&config)?;
+
+        let bytecode = fetch_creation_code(contract, client, provider).await?;
+
+        let args_arr = parse_constructor_args(bytecode, contract, &etherscan, abi_path).await?;
+        for arg in args_arr {
+            println!("{arg}");
```

Comment:
I think we should switch to from `println!` to `sh_println!` macro we introduced recently